### DISCUSSION TITLE
Completion in blade files

### DISCUSSION THREAD

**Discussion by muneebusmani** [phpactor/phpactor#2667] [General]
Is it possible to get completions in blade files in these :
{{ }}

these brackets can contain regular php code, but i couldn't find a way to get completions in these brackets in neovim

**Comment by dantleech** [comment]
Converted to a discussion - I'm not personally aware of a way to do this, but maybe some inspiration can be taken from Laravel plugins in the VS code ecosystem?

**Comment by muneebusmani** [comment]
vscode doesnt do it either, whenever i do blade files, i gotta use phpstorm for that because there is no way (according to my knowledge) to make lsp treat {{ echo("Hello World") }} as \<\?php echo("Hello World")  \?> , it is that under the hood but lsp cant recognize that

**Reply by mamazu** [reply]
How about language injections? Like for example if you write a string containing SQL the LSP it automatically highlights this as SQL. Maybe something similar like this is possible in blade as well. Depending on the LSP Syntax.

**Reply by muneebusmani** [reply]
i never talked about syntax highlighting, it's about making lsp server understand blade templating language, i couldn't get a way to make that work, i tried laravel-dev-tools but that guy i guess forgot his github password and didnt commit for a whole year even though his lsp is based on phpactor and it supports blade, it doesnt work on up to date laravel

**Comment by muneebusmani** [comment]
i just opened vscode and saw devsense updated, and they mentioned the intellisense for blade files, sadly that lsp is closed source, and i dont really know how to utilize vscode lsps in neovim

**Comment by dantleech** [comment]
note that @haringsrob started a POC Larvel extension which seems to support blade files: https://github.com/phpactor/phpactor/pull/1991 I'm not sure what challenges there are to support this in the main project, and sadly I don't have much time.

**Comment by muneebusmani** [comment]
i think he abandoned the project, because there are issues from last year and his last update is very old, and ik i am not saying that i want that in phpactor, this lsp is great btw for the rest of the php, as for the blade files, i gotta use vscode for them, which i rarely open because i started doing laravel + inertia(react), anyways keep up with the good work and sorry for doing some emotional comments a while ago, because i was new to neovim and you know how frustrated a person is when they just joined neovim, gotta close the discussion because there is no current solution yet.
